### PR TITLE
Autodetect identity region and other fixes

### PR DESCRIPTION
**Pull request by iakov-aws**
*Issue #, if available:*

*Description of changes:*
* Autodetect QS identity region in CFN
* Fix issues with Autodetect QS identity region in CID-CMD
* minor changes in automated testing of CID (just speed up)
* Fix issue with datasource listing (as it was breaking automated testing of CFN)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cid/test/bats/10-deploy-update-delete/compute-optimizer-dashboard.bats [hunk 1]
```diff
@@ -5,6 +5,7 @@ account_id=$(aws sts get-caller-identity --query "Account" --output text )
 @test "Install" {
   run cid-cmd -vv deploy  \
     --dashboard-id compute-optimizer-dashboard \
+    --share-with-account yes \
```
Comment (darken99) [inline cid/test/bats/10-deploy-update-delete/compute-optimizer-dashboard.bats:8]: We agreed to fix this

**Review by darken99 — CHANGES_REQUESTED** [review]

File: cid/test/bats/10-deploy-update-delete/trends-dashboard.bats [hunk 2]
```diff
@@ -9,6 +9,7 @@ database_name="${database_name:-athenacurcfn_cur1}" # If variable not set or nul
     --dashboard-id trends-dashboard \
     --athena-database $database_name\
     --account-map-source dummy \
+    --share-with-account yes \
```
Comment (darken99) [inline cid/test/bats/10-deploy-update-delete/trends-dashboard.bats:12]: Same

File: cid/test/bats/10-deploy-update-delete/ta-organizational-view.bats [hunk 3]
```diff
@@ -6,6 +6,8 @@ account_id=$(aws sts get-caller-identity --query "Account" --output text )
   run cid-cmd -vv deploy  \
     --dashboard-id ta-organizational-view \
     --athena-database 'optimization_data' \
+    --share-with-account yes \
```
Comment (darken99) [inline cid/test/bats/10-deploy-update-delete/ta-organizational-view.bats:9]: Same

File: cid/test/bats/10-deploy-update-delete/kpi_dashboard.bats [hunk 4]
```diff
@@ -9,6 +9,7 @@ database_name="${database_name:-athenacurcfn_cur1}" # If variable not set or nul
     --dashboard-id kpi_dashboard \
     --athena-database $database_name\
     --account-map-source dummy \
+    --share-with-account yes \
```
Comment (darken99) [inline cid/test/bats/10-deploy-update-delete/kpi_dashboard.bats:12]: Same

File: cid/test/bats/10-deploy-update-delete/cudos.bats [hunk 5]
```diff
@@ -11,6 +11,7 @@ quicksight_user="${quicksight_user:-cicd-staging}" # If variable not set or null
     --athena-database $database_name\
     --account-map-source dummy \
     --quicksight-user $quicksight_user \
+    --share-with-account yes \
```
Comment (darken99) [inline cid/test/bats/10-deploy-update-delete/cudos.bats:14]: Same

File: cid/test/bats/10-deploy-update-delete/cost_intelligence_dashboard.bats [hunk 6]
```diff
@@ -9,6 +9,7 @@ database_name="${database_name:-athenacurcfn_cur1}" # If variable not set or nul
     --dashboard-id cost_intelligence_dashboard \
     --athena-database $database_name\
     --account-map-source dummy \
+    --share-with-account yes \
```
Comment (darken99) [inline cid/test/bats/10-deploy-update-delete/cost_intelligence_dashboard.bats:12]: Same

**Review by darken99 — CHANGES_REQUESTED** [review]

File: cid/test/bats/10-deploy-update-delete/compute-optimizer-dashboard.bats (hunk 1 above)
Comment (iakov-aws) [reply cid/test/bats/10-deploy-update-delete/compute-optimizer-dashboard.bats]: fixed now

**Review by darken99 — APPROVED** [review]

**Pull request merged**